RevenueCatUI/PaywallFontProvider.swift

```diff
@@ -49,6 +49,10 @@ open class DefaultPaywallFontProvider: PaywallFontProvider {
         case .caption: return .caption
         case .caption2: return .caption2
 
+        #if compiler(>=6.0) && os(tvOS)
+        case .caption3: return .caption3
+        #endif
```

Comment:
This got added in iOs 18 but only for tvOS

https://developer.apple.com/documentation/swiftui/font/textstyle/caption3

Also need to use `compiler()` (the version of the compiler) instead of `swift()` (the swift mode used), since we want to check we're compiling in Xcode 16, regardless of the swift mode.